src/main/java/com/aws/greengrass/logmanager/util/ActiveFileCompletedListener.java

```diff
@@ -0,0 +1,22 @@
+package com.aws.greengrass.logmanager.util;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class ActiveFileCompletedListener {
```

Comment:
The listeners are actually just the consumers listening to events.

Also this should be directly part of the LogManagerService and it would just expose the registerListenerMethod as public

```suggestion
public class LogManagerService {
   private final List<Consumer<Event>> listeners = new ArrayList<>();

   static enum Event {
     LOG_GROUP_PROCESSED
   }

    public Runnable registerListener(Consumer<Event> c) {
         listeners.add(c);
         return () -> listeners.remove(c);
    }

   private void emit(Event e);
```

Also an event can ve anything, don't just make it be able to emit ActiveFile statuses. It doesn't work well of you then want to trigger other events.